test/unit/any_allocator.cpp

```diff
@@ -0,0 +1,699 @@
+        BOOST_TEST(alloc_t(b1) == a5);
+
+        // A a1(std::move(a)): Constructs a1 such
+        // that it equals the prior value of a
+        alloc_t a6;
+        alloc_t prior_a6(a6);
+        alloc_t a7(std::move(a6));
+        BOOST_TEST(a7 == prior_a6);
+
+        // A a1 = std::move(a): Constructs a1 such
+        // that it equals the prior value of a
+        alloc_t a8;
+        alloc_t prior_a8(a8);
+        alloc_t a9 = std::move(a8);
+        BOOST_TEST(a9 == prior_a8);
+
+        // A a(std::move(b)): Constructs a such
+        // that it equals the prior value of A(b)
+        any_allocator<int> b2;
+        alloc_t prior_ab(b2);
+        alloc_t a10(std::move(b2));
+        BOOST_TEST(a10 == prior_ab);
+    }
+#else
+    template <typename T, bool stateful>
+    static void
+    testDefaultConstruct()
+    {
+    }
+#endif
+
+    template <typename T, bool stateful>
+    static void
+    testVector()
+    {
+        using alloc_t = any_allocator<T>;
+
+        auto mem = get_resource<T, stateful>();
+        auto base = get_allocator<T, stateful>(
+            mem);
+        alloc_t any_alloc(base);
+
+        std::vector<T, alloc_t> v(any_alloc);
+        v.reserve(v.capacity() + 1);
+        v.reserve(v.capacity() + 1);
+        v.shrink_to_fit();
+        v.reserve(v.capacity() + 1);
+    }
+
+    template <typename T, bool stateful>
+    static void
+    testList()
+    {
+        using alloc_t = any_allocator<T>;
+
+        auto mem = get_resource<T, stateful>();
+        auto base = get_allocator<T, stateful>(
+            mem);
+        alloc_t any_alloc(base);
+
+        std::list<T, alloc_t> v(any_alloc);
+
+        v.push_back(get_value<T>());
+        v.push_back(get_value<T>());
+        v.push_back(get_value<T>());
+        v.pop_back();
+        v.pop_back();
+        v.pop_back();
+    }
+
+    template <typename T>
+    static void
+    testRequirements()
+    {
+        testTypes<T>();
+        testPointer<T>();
+        testRelationships<T, false>();
+        testRelationships<T, true>();
+        testLifetime<T, false>();
+        // testLifetime<T, true>();
+        testContainerOps<T, false>();
+        testContainerOps<T, true>();
+        testDefaultConstruct<T>();
+        testVector<T, false>();
+        testVector<T, true>();
+        testList<T, false>();
+        testList<T, true>();
+    }
+
+    template <std::size_t N>
+    static enable_if_t< alignof(max_align_t) <= N,
+        void>
+    testTryAlignedNamedRequirements()
+    {
+        testRequirements<
+            max_align_t>();
+    }
+
+    template <std::size_t N>
+        static enable_if_t <
+        N< alignof(max_align_t), void>
+        testTryAlignedNamedRequirements()
+    {
+        testRequirements<
+            type_with_alignment<2>::type>();
+        testTryAlignedNamedRequirements<N * 2>();
+    }
+
+    static void
+    testAligned()
+    {
+        testTryAlignedNamedRequirements<1>();
+    }
+
+    struct large_struct
```

Comment:
`struct S` lol

Reply:
LoL

The intention of this large struct is to test something whose size is more than max_align but still needs padding. I couldn't find a short name for that.

Yes. I think I'll just go with `S` then haha